app/controllers/forms_controller.rb

```diff
@@ -41,8 +41,19 @@ def calculate_early_result
     if last_tag_in_group?(:gross_income)
       cfe_result = CfeService.result(session_data, Steps::Helper.completed_steps_for(session_data, step))
       session_data["early_result"] = { "result" => cfe_result.gross_income_result,
-                                       "gross_income_excess" => cfe_result.gross_income_excess }
-      # add analytics tracking here? completed user journeys
+                                       "gross_income_excess" => cfe_result.gross_income_excess,
+                                       "type" => "gross_income" }
+      if @check.early_ineligible_result?
+        track_completed_journey_for_early_result
+      end
     end
   end
+
+  def track_completed_journey_for_early_result
+    session_data["api_response"] = CfeService.call(session_data, Steps::Helper.completed_steps_for(session_data, step))
+    calculation_result = CalculationResult.new(session_data)
+    office_code = signed_in? && current_provider.present? ? current_provider.first_office_code : nil
+    assessment_id_with_early_result_suffix = "#{assessment_id}_early_result" # This is so we do not overrite newly created early result entry, if an user decides to carry on with check or change answers
```

Comment:
with the addition of the new early result boolean field, we no longer plan to use the suffix to differentiate between early/full checks.

Instead we'll start allowing a check to be saved more than once i.e. take the unique constraint off the assessment_id column in the CompletedUserJourneys table.

Then the update action will change to use a combination of the assessment code and the early result boolean to know if it's updating an existing row, or creating a new one. So if it's an early check, the active record query should look to update an existing row for the assessment id and early_eligibility_result: true. If it's not an early check, it'll look for assessment id and early_eligibility_result: false. If neither of those exist it'll create the new row.

Reply:
I'm a bit confused. Should I be working on this ticket until that migration? It still seems blocked.

Reply:
So the plan that we agreed at dev time was that the migration would be done on this ticket as it is an integral part to saving the early result. Are you happy to do that?

Reply:
If for any reason it becomes complicated and adds too much complexity to this ticket (hopefully it is straight forward but that is always famous last words) then let us know and we can look at splitting the work out again